### PR TITLE
Fix for config file not being written

### PR DESCRIPTION
**Pull request by gareth-j**
This fixes an issue with the configuration file not being written out correctly. It also adds a quick `check_config` function that makes sure the file is in the correct format and contains the correct items.

### CODE REVIEW

**Comment by gareth-j** [comment]
Closes #566 and #565

File: openghg/util/_user.py [hunk 1]
```diff
@@ -153,3 +146,30 @@ def read_local_config() -> Dict:
...
+
+    config = read_local_config()
+    uid = config["user_id"]
+    object_stores = config["object_store"]
+
+    try:
+        uuid.UUID(uid, version=4)
+    except ValueError:
+        logger.exception("Invalid user ID. Please re-run quickstart to setup a valid config file.")
+        raise
```
Comment (rt17603) [inline openghg/util/_user.py:171]: Was this meant to raise a specific error here?

File: openghg/util/_user.py [hunk 2]
```diff
@@ -153,3 +146,30 @@ def read_local_config() -> Dict:
...
+    """Check that the user config file is valid and the paths
+    given in it exist.
+
+    Returns:
+        bool
+    """
+    config_path = get_user_config_path()
+
+    if not config_path.exists():
+        logger.warning("Configuration file does not exist. Please create it by running openghg --quickstart.")
```
Comment (rt17603) [inline openghg/util/_user.py:161]: Does this mean we would now expect the user to always run `$ openghg --quickstart` on the command line?

File: .github/workflows/workflow.yaml [hunk 3]
```diff
@@ -46,7 +46,7 @@ jobs:
     - name: Test with pytest
       run: |
         pip install --upgrade pytest
-        pytest --run-cfchecks --cov=./ --cov-report=xml
+        pytest -s --run-cfchecks --cov=./ --cov-report=xml
```
Comment (rt17603) [inline .github/workflows/workflow.yaml:49]: Do we still need `-s` in the pytest checks or was this just needed for checking the issue this PR is fixing?

**Review by rt17603 — COMMENTED** [review]
Would be good to add a description to this PR to make it clear what is being added/changed. A few comments/questions added here.

File: openghg/util/_user.py (hunk 2 above)
Comment (gareth-j) [reply openghg/util/_user.py:161]: Ah good point, I'll update that so it also asks them to call `create_config`. I'll check the setup documentation as well.

File: .github/workflows/workflow.yaml (hunk 3 above)
Comment (gareth-j) [reply .github/workflows/workflow.yaml]: Yeah there was an issue with the capture of output so I added that, I've removed it now and everything works.

Comment (rt17603) [reply .github/workflows/workflow.yaml]: Great, makes sense.

**Comment by rt17603** [comment]
By the way, I *think* if you updated your PR description to include the "Closes #566 and #565" this will automatically close these issues when this is merged. Otherwise, I don't think it will.

File: openghg/util/_user.py (hunk 1 above)
Comment (gareth-j) [reply openghg/util/_user.py:171]: Because it's in the except of a ValueError, it's so you can do something before re-raising the exception.

Comment (rt17603) [reply openghg/util/_user.py:171]: Ah that's useful to know and I've learnt something. Thanks for explaining.

**Review by rt17603 — APPROVED** [review]
Looks good. The only question is how we want to handle running the tutorials and whether we want to allow this to bypass the create config step but this can be addressed on a separate PR.

**Pull request merged**